distributed/protocol/arrow.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import print_function, division, absolute_import
+
+from .serialize import register_serialization
+
+
+def serialize_batch(batch):
+    import pyarrow as pa
+    sink = pa.BufferOutputStream()
+    writer = pa.RecordBatchStreamWriter(sink, batch.schema)
+    writer.write_batch(batch)
+    writer.close()
+    buf = sink.get_result()
+    header = {}
+    frames = [buf.to_pybytes()]
```

Comment:
This causes an extra memory copy. Can `frames` contain objects exporting the buffer protocol?

Reply:
That's a good question.  It looks like it can with small modification (pushed).  Presumably this means that PyArrow also works with sockets and Tornado IOStreams.  Hooray for consistent use of protocols.